pkg/azurediskplugin/dev.Dockerfile

```diff
@@ -20,3 +20,4 @@ ARG ARCH=amd64
 ARG PLUGIN_NAME=azurediskplugin
 COPY ./_output/${ARCH}/${PLUGIN_NAME} /azurediskplugin
 ENTRYPOINT ["/azurediskplugin"]
+
```

Comment:
This file has been removed from upstream. We should remove it as well.

Reply:
As discussed, we will continue referencing this file in Makefile for v2?

Reply:
See my [previous comment](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/pull/205#discussion_r719526284).

Reply:
Removed the file. Everything refers to Dockerfile now.